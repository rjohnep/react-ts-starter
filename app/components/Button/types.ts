import { ReactNode } from 'react';

import { Variants as ButtonVariantsE } from '@app/theme';

export { ButtonVariantsE };

export type ButtonPropsT = {
  variant?: ButtonVariantsE;
  children?: ReactNode;
};
