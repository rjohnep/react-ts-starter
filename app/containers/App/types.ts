export type UserT = {
  firstName: string;
  lastName: string;
  gender?: string;
  years?: string;
  roles?: string[];
};
