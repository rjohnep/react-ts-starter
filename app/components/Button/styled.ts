import styled, { css } from 'styled-components';

import { ButtonVariantsE } from './types';

const primaryStyle = css`
  color: ${(props) => props.theme.colors[ButtonVariantsE.Primary]};
  background: ${(props) => props.theme.colors[ButtonVariantsE.Secondary]};
`;

const secondaryStyle = css`
  color: ${(props) => props.theme.colors[ButtonVariantsE.Secondary]};
  background: ${(props) => props.theme.colors[ButtonVariantsE.Primary]};
`;

const successStyle = css`
  color: ${(props) => props.theme.colors[ButtonVariantsE.Success]};
  background: ${(props) => props.theme.colors[ButtonVariantsE.Secondary]};
`;

export const Wrapper = styled.div<{ variant: ButtonVariantsE }>`
  ${({ variant }) => {
    switch (variant) {
      case ButtonVariantsE.Secondary:
        return secondaryStyle;
      case ButtonVariantsE.Success:
        return successStyle;
      case ButtonVariantsE.Primary:
      default:
        return primaryStyle;
    }
  }};

  border-radius: 4px;
`;
