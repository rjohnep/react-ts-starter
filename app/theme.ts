import baseStyled, { ThemedStyledInterface } from 'styled-components';

export enum Variants {
  Primary = 'Primary',
  Secondary = 'Secondary',
  Success = 'Success'
}

const colors = {
  [Variants.Primary]: '#fff',
  [Variants.Secondary]: '#000',
  [Variants.Success]: '#3cb371'
};

export const theme = {
  colors
};

export type ThemeT = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<ThemeT>;
