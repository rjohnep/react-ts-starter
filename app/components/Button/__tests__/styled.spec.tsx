import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import { theme } from '@app/theme';
import { Wrapper } from '../styled';
import { ButtonVariantsE } from '../types';

const makeButtonVariantSnap = (
  variant: ButtonVariantsE
): ReactTestRendererJSON | null => renderer.create(
  <ThemeProvider theme={theme}>
    <Wrapper variant={variant} />
  </ThemeProvider>
).toJSON();

describe('<Button />', () => {
  it('should render primary variant', () => {
    const renderedOutput = makeButtonVariantSnap(ButtonVariantsE.Primary);

    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render secondary variant', () => {
    const renderedOutput = makeButtonVariantSnap(ButtonVariantsE.Secondary);

    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render success variant', () => {
    const renderedOutput = makeButtonVariantSnap(ButtonVariantsE.Success);

    expect(renderedOutput).toMatchSnapshot();
  });
});
