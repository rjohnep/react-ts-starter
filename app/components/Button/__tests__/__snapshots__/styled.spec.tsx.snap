// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render primary variant 1`] = `
.c0 {
  color: #fff;
  background: #000;
  border-radius: 4px;
}

<div
  className="c0"
/>
`;

exports[`<Button /> should render secondary variant 1`] = `
.c0 {
  color: #000;
  background: #fff;
  border-radius: 4px;
}

<div
  className="c0"
/>
`;

exports[`<Button /> should render success variant 1`] = `
.c0 {
  color: #3cb371;
  background: #000;
  border-radius: 4px;
}

<div
  className="c0"
/>
`;
